tests/components/tcp/test_sensor.py

```diff
@@ -244,3 +245,29 @@ async def test_ssl_state_verify_off(
     )
     assert mock_ssl_socket.recv.called
     assert mock_ssl_socket.recv.call_args == call(SENSOR_TEST_CONFIG["buffer_size"])
+
+
+async def test_update_socket_error_after_success(
+    hass: HomeAssistant, mock_socket
+) -> None:
+    """Test state is updated with socket errors during update, when there is an old state."""
+    assert await async_setup_component(hass, "sensor", TEST_CONFIG)
+    await hass.async_block_till_done()
+
+    state = hass.states.get(TEST_ENTITY)
+
+    assert state
+    assert state.state == "7.123"
+
+    mock_socket.connect.side_effect = OSError("Boom")
+    # FIXME: where is the default scan interval for tcp sensor?
```

Comment:
We prefer to use `freezer.tick` to move time. Besides that no need to use a constant. I always just set 60 seconds as that's the max any platform use.

Reply:
I'll take a look at freezer.tick thanks!

Regarding the constant, I understand your point but it sounds like a bad approach towards maintainability

Default is set somewhere for sure, I can reuse that value rather than repeating it and avoiding updating it in multiple places if it changes

Reply:
`from homeassistant.components.sensor import SCAN_INTERVAL`
Unless you have defined something on your own in the integration but I guess not given the question